File: Design_Pattern_and_Principles/DependencyInjectionExample/DITest.java
public class DITest {
    public static void main(String[] args) {
        CustomerRepository repoImpl = new CustomerRepositoryImpl();

        CustomerService service = new CustomerService(repoImpl);

        System.out.println("Processing execution pipeline query check:");
        service.lookupCustomer("CUST-7741");
    }
}

File: Design_Pattern_and_Principles/DependencyInjectionExample/DITest.cs
using System;

public class DITest {
    public static void Main(string[] args) {
        ICustomerRepository repoImpl = new CustomerRepositoryImpl();
        CustomerService service = new CustomerService(repoImpl);

        Console.WriteLine("Processing execution pipeline query check:");
        service.LookupCustomer("CUST-7741");
    }
}